app/controllers/vitals_controller.rb

```diff
@@ -1,6 +1,6 @@
 class VitalsController < ApplicationController
   before_action :set_vital, only: %i[ show edit update destroy ]
-  before_action { authorize(@vital || Vital) }
+  before_action :authorize_resource, except: %i[chart]
```

Comment:
There's no real reason to create a separate method called authorize_resource if the sole purpose is to have an exception for the authorize. Instead I believe you can do something like 
```
before_action(except: [:chart]) { authorize(@vital || Vital) }
```
to specify the exception